metadata.go

```diff
@@ -200,15 +212,34 @@ func GetImageToMDMap(imageMDs []ImageMetadataInfo) (imageToMDMap map[string][]Im
 	return
 }
 
-// getLocalImages queries the local Docker daemon for list of images.
-func getLocalImages() (imageMap map[ImageIDType][]RepoTagType, e error) {
+// CheckRepoToProcess returns true if ReposToProcess is empty
+// or if the given repo string ends with one of the repo names in ReposToProcess.
+// The 2nd condition is needed to handle local image metadata including registry info in repository info.
+func CheckRepoToProcess(repo string) bool {
+	if len(ReposToProcess) == 0 {
+		return true
+	}
+
+	for keyRepo, valBool := range(ReposToProcess) {
```

Comment:
May be better to detect if repo starts with a registry name. Strip off the registry name and do a single map lookup on the remainder, which is just the repo path.